src/main/java/io/stargate/sgv2/jsonapi/api/model/command/clause/filter/LogicalExpression.java

```diff
@@ -26,13 +26,47 @@ public String getOperator() {
     }
   }
 
+  // TODO: why are these not final ?
+  public List<LogicalExpression> logicalExpressions;
+  public List<ComparisonExpression> comparisonExpressions;
+
   private LogicalOperator logicalRelation;
+  // TODO - there MUST be explaination of why this count is needed, why not just use the size of the
```

Comment:
I know this is just first of PRs so possibly hard to change, but I think the answer is that due to nesting of expressions, root-level lists do not tell the full answer (needs to recursively descend to calculate). But should have Javadoc to state that.

cc @Yuqi-Du who knows this better than I do.

Reply:
yes, like Tatu said, expressions are nested, this is counter lives in the root-level that sums up all the expression in the tree.

Reply:
added doc and TODO